### PR TITLE
feat(mypy) Type hint snuba.cli

### PR DESCRIPTION
**Pull request by evanh**
Also fix --strict errors in snuba.query

### CODE REVIEW

File: snuba/cli/subscriptions.py [hunk 1]
```diff
@@ -119,9 +119,12 @@ def subscriptions(
     dataset = get_dataset(dataset_name)
 
     storage = dataset.get_default_entity().get_writable_storage()
+    assert storage is not None
     storage_key = storage.get_storage_key()
 
     loader = enforce_table_writer(dataset).get_stream_loader()
+    topic_spec = loader.get_commit_log_topic_spec()
+    assert topic_spec is not None
```
Comment (fpacifici) [inline snuba/cli/subscriptions.py:127]: All datasets are supposed to have a writable storage, we should be able to make that non nullable. I think we should fix it there and leave this as it is. If we add the assertion here we are hiding the issue.

File: snuba/query/parser/expressions.py [hunk 2]
```diff
@@ -77,7 +77,7 @@
 )
 
 
-class ClickhouseVisitor(NodeVisitor):
+class ClickhouseVisitor(NodeVisitor):  # type: ignore
```
Comment (fpacifici) [inline snuba/query/parser/expressions.py:80]: What was the issue here ?

Comment (evanh) [reply snuba/query/parser/expressions.py:80]: NodeVisitor has type Any and mypy warns on extending something with type: Any.

File: snuba/cli/subscriptions.py (hunk 1 above)
Comment (evanh) [reply snuba/cli/subscriptions.py:127]: Just to clarify, this is a field on the KafkaStreamLoader, not the loader itself. And we actually pass None into that field sometimes: https://github.com/getsentry/snuba/blob/master/snuba/datasets/table_storage.py#L122

File: snuba/query/parser/expressions.py (hunk 2 above)
Comment (fpacifici) [reply snuba/query/parser/expressions.py:80]: Mind adding a comment to explain ?

File: snuba/cli/subscriptions.py (hunk 1 above)
Comment (fpacifici) [reply snuba/cli/subscriptions.py:127]: you are right on this, but this comment also addresses line 122. Every dataset should have a writable storage.

**Review by fpacifici — APPROVED** [review]

**Pull request merged**